PyInstaller/lib/modulegraph/modulegraph.py

```diff
@@ -2608,200 +2636,63 @@ def _scan_ast(self, module, module_code_object_ast):
-        #
-        #No other changes are required, which is relieving to us all.
-
-        def get_operation_arg():
-            """
-            Get the argument passed to the current bytecode operation as an
-            unsigned short (i.e., 2-byte integer).
-            """
-            return unpack(
-                '<H', code_bytes[code_byte_index - 2:code_byte_index])[0]
-
-        def get_operation_arg_name():
-            """
-            Get the name from this module's code object whose index in this
-            list of names is the unsigned short (i.e., 2-byte integer) argument
-            passed to the current bytecode operation.
-            """
-            co_names_index = get_operation_arg()
-            return module_code_object.co_names[co_names_index]
-
-        # For each byte index into this list of bytecode bytes...
-        while code_byte_index < num_code_bytes:
-            # Opcode signifying the current type of operation being performed.
-            code_byte = code_bytes[code_byte_index]
-
-            # Iterate the current byte index past this 1-byte opcode.
-            code_byte_index += 1
-
-            # If this opcode accepts an argument, this argument is guaranteed
-            # by design to be two bytes in size. As the current byte index is
-            # at the first of these two bytes, iterating this index past both
-            # bytes guarantees this index to now point at another opcode.
-            #
-            # To obtain the index of the first byte of this argument, simply
-            # decrement "2" from this index (e.g., "code_byte_index - 2").
-            #
-            # Note, however, that this brute-force approach effectively ignores
-            # extended arguments. While most opcodes accepting an argument only
-            # require a 2-byte argument, some require a 4-byte argument. To
-            # seamlessly support both, the most significant two bytes of such
-            # 4-byte arguments are split into a preceding "EXTENDED_ARG"
-            # opcode. For the moment, no opcode below requires such arguments.
-            #
-            # For further details, see:
-            #     https://docs.python.org/3/library/dis.html#opcode-HAVE_ARGUMENT
-            #     https://docs.python.org/3/library/dis.html#opcode-EXTENDED_ARG
-            if code_byte >= _HAVE_ARGUMENT_OPCODE:
-                code_byte_index = code_byte_index+2
-
-            # If this is an import statement originating from this module,
-            # parse this import.
-            #
-            # Note that the related "IMPORT_FROM" opcode need *NOT* be parsed.
-            # "IMPORT_NAME" suffices. For further details, see
-            #     http://probablyprogramming.com/2008/04/14/python-import_name
-            if code_byte == _IMPORT_NAME_OPCODE:
+        level = None
+        fromlist = None
+
+        all_instructions = dis.get_instructions(module_code_object)
```

Comment:
This creates a list of all Instructions. The dis-module has a nice iterator interface, let's use it!

For the cases we need to access earlier instructions, a `deque([], maxize=2)` should do the trick. This will keep the last 2 instructions. Depending on what information we need and what we want to test, we could even stack e.g. `(inst.opname, inst.argval)` tuples there, or use two deques.